### ISSUE TITLE
Proposal: no linebreak after `get`, `set`, `static`

### ISSUE THREAD

**Issue by bakkot** [eslint/eslint#11261]
**Please describe what the rule should do:**

Forbid linebreaks after `get`, `set`, and `static` modifiers in class bodies and (except `static`) object literals. (It is already illegal to have a linebreak after `async`.)

**What category of rule is this? (place an "X" next to just one item)**

[ ] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ x ] Enforces code style (layout)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```js
class A {
  get
    x(){}
}
```

```js
let obj = {
  set
    x(y){}
};
```

```js
class A {
  static
    x(){}
}
```

**Why should this rule be included in ESLint (instead of a plugin)?**

It is applicable to anyone writing ES2015 classes or object literals. I've never seen any nontrivial codebase use the style it forbids - possibly because, until fairly recently, neither [babel](https://github.com/babel/babylon/pull/351) nor [typescript](https://github.com/Microsoft/TypeScript/issues/22757) parsed these correctly. But it is legal, and the rule needs to be enforced by anyone trying to enforce consistent style.

Also, if the [class fields proposal](https://github.com/tc39/proposal-class-fields/) goes forward in its current state, it would make the following legal:

```js
class A {
  async
    x(){}
}
```

as an uninitialized public field named `async` followed by a method. That inconsistency would make the code forbidden by this proposed rule especially confusing.

**Are you willing to submit a pull request to implement this rule?**

Yes.
